### PR TITLE
fix: use dataLength

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
Minor bug when passing strings, we shouldn't look at the string char length but it's byte length

File: lib/sender.js [hunk 1]
```diff
@@ -135,6 +135,10 @@ class Sender {
 
     if (skipMasking) return [target, data];
 
+    if (typeof data === 'string') {
+      data = Buffer.from(data);
+    }
```
Comment (ronag) [inline lib/sender.js:140]: This isn't strictly need but makes things safer for bugs introduced by future changes.

File: lib/sender.js [hunk 2]
```diff
@@ -103,10 +103,10 @@ class Sender {
 
     let payloadLength = dataLength;
 
-    if (data.length >= 65536) {
+    if (dataLength >= 65536) {
```
Comment (lpinca) [inline lib/sender.js:106]: Good catch. Thanks.

File: lib/sender.js (hunk 1 above)
Comment (lpinca) [reply lib/sender.js:140]: Not very happy with this. I would prefer to use `data.length` below.

**Review by lpinca — APPROVED** [review]

**Comment by lpinca** [comment]
Thank you. I will merge this and cut a new release later.

**Pull request merged**